apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -1557,36 +1564,41 @@ impl FederatedQueryGraphBuilder {
                     .or_default()
                     .push(object_field_arg.clone());
                 let field_coordinate = object_field_arg.parent();
-                if let Some(dir) = input_value.directives.get(
+                let Some(dir) = input_value.directives.get(
                     subgraph_data
                         .from_context_directive_definition_name
                         .as_str(),
-                ) {
-                    let application = subgraph_data
-                        .federation_spec_definition
-                        .from_context_directive_arguments(dir)?;
-                    let (context, selection) = parse_context(application.field)?;
-
-                    let types_with_context_set = context_name_to_types
```

Comment:
Bugs:
- There should always be an entry in `context_name_to_types` for the context name, if not we should `bail!()` instead of silently ignore it.